File: solutions/lua/all-your-base/1/all-your-base.lua
local all_your_base = {}
all_your_base.convert = function(from_digits, from_base)
  local result = 0
  local self = {}
  for i = 1, #from_digits, 1 do
    local tmp = from_digits[#from_digits - (i - 1)]
    result = result + (tmp * math.pow(from_base, (i - 1)))
  end
  local function reverse(arr)
    for i = 1, #arr / 2, 1 do
      local tmp = arr[i]
      arr[i] = arr[#arr - (i - 1)]
      arr[#arr - (i - 1)] = tmp
    end
    return arr
  end
  function self.to(base)
    local tmp = {}
    tmp[1] = 0
    local ii = 1
    while (result > 0) do
      tmp[ii] = math.floor(result % base)
      result = math.floor(result / base)
      ii = ii + 1
    end
    return reverse(tmp)
  end
  return self
end

return all_your_base

File: solutions/go/all-your-base/1/all_your_base.go
package allyourbase

import (
	"errors"
	"math"
)

func ConvertToBase(inputBase int, inputDigits []int, outputBase int) ([]int, error) {
	if len(inputDigits) == 0 && inputBase >= 2 {
		return []int{0}, errors.New("")
	}
	if inputBase < 2 {
		return []int{0}, errors.New("input base must be >= 2")
	}
	if outputBase < 2 {
		return []int{0}, errors.New("output base must be >= 2")
	}

	base10 := 0

	index := len(inputDigits) - 1
	for i := range inputDigits {
		tmp := inputDigits[index]
		if tmp < 0 || tmp >= inputBase {
			return []int{0}, errors.New("all digits must satisfy 0 <= d < input base")
		}
		base10 += inputDigits[index] * int(math.Pow(float64(inputBase), float64(i)))
		index--
	}

	result := []int{}
	if base10 == 0 {
		return []int{0}, nil
	}
	for i := base10; i > 0 || i == outputBase; i /= outputBase {
		result = append([]int{(i % outputBase)}, result...)
	}
	return result, nil
}
